lib/presentation/widget/upload_request_creation/upload_request_creation_viewmodel.dart

```diff
@@ -84,12 +140,270 @@ class UploadRequestCreationViewModel extends BaseViewModel {
+        _maxExpirationDateRoundUp = textActivationNotifier.value!.value1.add(Duration(days: _expirationSetting!.maxValue * 7));
+        break;
+      case UnitTimeType.MONTH:
+        _initExpirationDateRoundUp = textActivationNotifier.value!.value1.copyWith(month: textActivationNotifier.value!.value1.month + _expirationSetting!.value);
+        _maxExpirationDateRoundUp = textActivationNotifier.value!.value1.copyWith(month: _creationDateRoundUp.month + _expirationSetting!.maxValue);
+        break;
+      case null:
+        _initExpirationDateRoundUp = textActivationNotifier.value!.value1;
+        _maxExpirationDateRoundUp = textActivationNotifier.value!.value1;
+        break;
+    }
+    textExpirationNotifier.value = Tuple2(_initExpirationDateRoundUp, _initExpirationDateRoundUp.getYMMMMdFormatWithJm());
+  }
+
+  void _getReminderDate() {
+    switch(_notificationSetting?.unit.toUnitTimeType()) {
+      case UnitTimeType.DAY:
+        _initReminderDateRoundUp = textExpirationNotifier.value!.value1.subtract(Duration(days: _notificationSetting!.value));
+        _minReminderDateRoundUp = textExpirationNotifier.value!.value1.subtract(Duration(days: _notificationSetting!.maxValue));
+        break;
+      case UnitTimeType.WEEK:
+        _initReminderDateRoundUp = textExpirationNotifier.value!.value1.subtract(Duration(days: _notificationSetting!.value * 7));
+        _minReminderDateRoundUp = textExpirationNotifier.value!.value1.subtract(Duration(days: _notificationSetting!.maxValue * 7));
+        break;
+      case UnitTimeType.MONTH:
+        _initReminderDateRoundUp = textExpirationNotifier.value!.value1.copyWith(month: textExpirationNotifier.value!.value1.month - _notificationSetting!.value);
+        _minReminderDateRoundUp = textExpirationNotifier.value!.value1.copyWith(month: textExpirationNotifier.value!.value1.month - _notificationSetting!.maxValue);
+        break;
+      case null:
+        _initReminderDateRoundUp = textExpirationNotifier.value!.value1;
+        _minReminderDateRoundUp = textExpirationNotifier.value!.value1;
+        break;
+    }
+    if(_minReminderDateRoundUp.compareTo(textActivationNotifier.value!.value1) < 0) {
+      _minReminderDateRoundUp = textActivationNotifier.value!.value1;
+    }
+    if(_initReminderDateRoundUp.compareTo(textActivationNotifier.value!.value1) < 0) {
+      _initReminderDateRoundUp = textActivationNotifier.value!.value1;
+    }
+    textReminderNotifier.value = Tuple2(_initReminderDateRoundUp, _initReminderDateRoundUp.getYMMMMdFormatWithJm());
+  }
+
+  void _initDefaultData() {
+    maxFileSizeTypeNotifier.value = _maxFileSizeSetting?.unit.toFileSizeType() ?? FileSizeType.GB;
+    totalFileSizeTypeNotifier.value = _totalFileSizeSetting?.unit.toFileSizeType() ?? FileSizeType.GB;
+    notificationLanguageNotifier.value = _notificationLanguageSetting?.value.toNotificationLanguage() ?? NotificationLanguage.FRENCH;
```

Comment:
why not `English`?

Reply:
I don't know, the default on creation is like that.